api/segments/views.py

```diff
@@ -31,21 +32,41 @@
                 "Optionally provide the id of an identity to get only the segments they match",
                 required=False,
                 type=openapi.TYPE_INTEGER,
-            )
+            ),
+            openapi.Parameter(
+                "q",
```

Comment:
Why are we not calling it `q` instead of `name`?

Reply:
I was trying to be consistent with other endpoints which have a search parameter. Although I think those are split between `q` and `search`. I think `name`, however, would imply an exact match?

Reply:
>  I think name, however, would imply an exact match?
Ah, yes good point.